src/main/java/org/pedrocarlos/core/FileTransformerExecutor.java

```diff
@@ -46,34 +51,145 @@ public FileTransformerExecutor(List<TransformerOperation> transformerOperations,
+        var transformers = findTransformer(options.operations(), options.inputType());
         var transformer = createTransformerPipeline(transformers);
-        pipe(options.input(), options.output(), transformer);
+
+        try {
+            pipe(options.input(), options.output(), options.threads(), transformer);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
     }
 
 
     /**
      * Applies the given UnaryOperator to each line of the input file, and writes the transformed lines to the output file.
      *
-     * @param input The input file.
-     * @param output The output file.
-     * @param function The UnaryOperator to apply to each line of the input file.
+     * @param input           The input file.
+     * @param output          The output file.
+     * @param numberOfThreads number of threads
+     * @param transformerFunction   The UnaryOperator to transform  each line of the input file.
      * @throws IOException if an I/O error occurs.
      */
-    private void pipe(File input, File output, UnaryOperator<String> function) throws IOException {
-        try (BufferedReader reader = readAndWriteStreamFactory.getStreamReader(input)) {
-            try (PrintStream writer = readAndWriteStreamFactory.getStreamWriter(output)) {
-                reader.lines().map(function).forEach(writer::println);
+    private void pipe(File input, File output, int numberOfThreads, UnaryOperator<String> transformerFunction) throws IOException, InterruptedException {
+
+        /*
+         * Splits the input file into chunks for each thread, processes the chunks in parallel, and stores the output of each
+         * chunk in a temporary file. Once all chunks have been processed, the temporary files are streamed in order to the final output.
+         */
+
+
+        var chunksDelimiters = getChunksDelimiters(numberOfThreads, input);
+
+        var temporalFiles = new ArrayList<File>();
+
+        /* Process chunks in parallel */
+        var executor = Executors.newFixedThreadPool(numberOfThreads);
+        for (var chunkDelimiter : chunksDelimiters) {
+            final var temporalFile = File.createTempFile("temporal-transformed-chunk-thread-" + chunkDelimiter.start(), "");
+            temporalFiles.add(temporalFile);
+            executor.execute(() -> pipeChunk(input, chunkDelimiter, temporalFile, transformerFunction));
+
+        }
+        executor.shutdown();
+        executor.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
+
+        /* Join all temporal files and build the final output */
+        try (var writer = readAndWriteStreamFactory.getStreamWriter(output)) {
+
+            var firstLineProcessed= new AtomicBoolean(true); // boolean to
+            for (var temporalFile : temporalFiles) {
+                try (var lines = Files.lines(temporalFile.toPath())) {
+                    lines.forEach(line -> {
+                        if(!firstLineProcessed.get()) {
```

Comment:
This condition it is necessary to prevent write an extra line in the output file.